installer/win/WixSharpInstaller/ProgressDialog.cs

```diff
@@ -0,0 +1,218 @@
+                                   messageRecord[1]    "RegisterUser"
+                                   messageRecord[2]    "Registering user"
+                                   messageRecord[3]    "[1]"
+
+                                */
+                                if (messageRecord.FieldCount >= 3)
+                                {
+                                    message = messageRecord[2].ToString();
+                                }
+                            }
+                            else
+                            {
+                                message = messageRecord.FormatString;
+                                if (message.IsNotEmpty())
+                                {
+                                    for (int i = 1; i < messageRecord.FieldCount; i++)
+                                    {
+                                        message = message.Replace("[" + i + "]", messageRecord[i].ToString());
+                                    }
+                                }
+                                else
+                                {
+                                    message = messageRecord[messageRecord.FieldCount - 1].ToString();
+                                }
+                            }
+
+                            if (message.IsNotEmpty())
+                                currentAction.Text = currentActionLabel.Text + " " + message;
+                        }
+                        catch { }
+                    }
+                    break;
+            }
+            return MessageResult.OK;
+        }
+
+        /// <summary>
+        /// Called when MSI execution progress is changed.
+        /// </summary>
+        /// <param name="progressPercentage">The progress percentage.</param>
+        public override void OnProgress(int progressPercentage)
+        {
+            progress.Value = progressPercentage;
+
+            if (progressPercentage > 0)
+            {
+                waitPrompt.Visible = false;
+            }
+        }
+
+        /// <summary>
+        /// Called when MSI execution is complete.
+        /// </summary>
+        public override void OnExecuteComplete()
+        {
+            currentAction.Text = null;
+            if (showingCancelConfirmation) 
+            {
+                complete = true;
+            }
+            else 
+            {
+                Shell.GoNext();
+            }
+        }
+
+        /// <summary>
+        /// Handles the Click event of the cancel control.
+        /// </summary>
+        /// <param name="sender">The source of the event.</param>
+        /// <param name="e">The <see cref="System.EventArgs" /> instance containing the event data.</param>
+        void cancel_Click(object sender, EventArgs e)
+        {
+            showingCancelConfirmation = true;
+            if (System.Windows.Forms.MessageBox.Show("Are you sure you want to cancel Symphony installation?",
```

Comment:
we should get this string from a strings file for internationalisation support in the future